template/contracts/src/BuyMeACoffee.sol

```diff
@@ -143,8 +159,27 @@ contract BuyMeACoffee {
     /**
      * @dev Function to get the memos
      */
-    function getMemos() public view returns (Memo[] memory) {
-        return memos;
+    function getMemos(uint256 index, uint256 size) public view returns (Memo[] memory) {
```

Comment:
@Sneh1999 @awilliams1-cb curios to hear thoughts on this approach.

Reply:
I dont think paginating inside the smart contract is a general pattern used, @awilliams1-cb feel free to correct me here though

Reply:
Typically, I would just iterate event logs, but that is not a very efficient mechanism across a large blockspan (I think most providers will prevent you going back too far, 1000 blocks maybe?).  Agree in a production environment we would be indexing these events and putting them in a low-latency data store.

Reply:
It's definitely more common to have pagination for non-view functions. In general, I would say the way people handle data like this is to watch for the corresponding event, then save that data as it streams in (or use a subgraph / other data aggregation tool).

For our use case I think it's fine to paginate this view function though

Reply:
makes sense, I remember @Zizzamia  wanted to ad to add `the graph` to boat, maybe it can be used here. We could also merge this PR in the meantime and add graph later whatever you guys think is the best

Reply:
i am ok to merge this PR now, and than as follow up we start working on using `the graph`.